airflow/utils/log/arbitrary_log_shipper.py

```diff
@@ -0,0 +1,83 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from airflow.compat.functools import cached_property
+from airflow.config_templates.airflow_local_settings import ARBITRARY_LOG_SHIPPING_ENABLED
+
+if TYPE_CHECKING:
+    from airflow.models.taskinstance import TaskInstance
+
+
+class ArbitraryLogShipper:
```

Comment:
Maybe this class should have an interface more like `log` e.g. maybe it should have methods `info` `debug` `warning` etc and handle accordingly?

Another alternative name to consider along with `TaskLogShipper` would be `TaskExceptionLogger`

Reply:
perhaps, we can enhance this later if this is not a hard requirement for making it a log-like interface 🙂 

Have kept the class name TaskLogShipper for now.